lib/boat.rb

```diff
@@ -0,0 +1,20 @@
+class Boat
+
+  attr_reader :coords_hash
+
+  def initialize(coords_array)
+    @coords_hash = Hash.new
+    coords_array.each do |coords|
+      @coords_hash[coords] = false
+    end
+    #Add code to throw an error if the boat_coords parameter given is not an array of arrays?
```

Comment:
Not sure I would prioritise adding this kind of validation. I would rely on the creation code to provide the required data type, otherwise you're kind of inventing a type validation system, so where to stop?